src/front/glsl/parser/declarations.rs

```diff
@@ -199,8 +199,16 @@ impl<'source> ParsingContext<'source> {
             // TODO: Should we try to make constants here?
             // This is mostly a hack because we don't yet support adding
             // bodies to entry points for variable initialization
-            let maybe_constant =
-                init.and_then(|(root, meta)| parser.solve_constant(ctx.ctx, root, meta).ok());
+            let maybe_constant = if let Some((root, meta)) = init {
+                let result = parser.solve_constant(ctx.ctx, root, meta);
+                match (result, ctx.external) {
```

Comment:
This looks correct, but I think it might be clearer as a plain `match result`, with a guard expression on the `Err` case:

```
Err(err) if ctx.external => return Err(err),
```

It also deserves a little comment, something like:

Initializers for external declarations must be constants, so if `solve_constant` failed, that's an error the user should see. But if it's an initializer for a local variable, then failing to produce a constant just means we emit the expression as code; that is not an error.

Reply:
Thank you very much for the review! I'd like to use what you wrote as a comment. :)

Reply:
Currently `maybe_constant` assigns the returned value of `sove_constant` for both externally and internally declared math functions if implemented. 

So, should it also be
```
Ok(res) if ctx.external => Some(res)
```
so that internally declared math expressions are left as is?